src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -231,7 +227,15 @@ private int tickAllGenerators(@Nonnull LongConsumer timings) {
                         loc.getWorld().createExplosion(loc, 0F, false);
                     });
                 } else {
-                    supply += energy;
+                    supply += provider.getGeneratedOutput(loc, data);
+
+                    if (provider.isChargeable()) {
+                        supply += provider.getCharge(loc, data);
+                    }
+                    
+                    if (supply < 0 || supply > MAX_ENERGY) {
+                        supply = MAX_ENERGY;
+                    }
```

Comment:
Congratulations, you broke reactors.

Reply:
forgot that getGeneratedOutput ran other code xD